### PR TITLE
Adds parquet2

### PR DESCRIPTION
**Pull request by jorgecarleitao**
This PR is pending IP clearance. Do not merge until IP is cleared.

It contains the source code of parquet2 rebased on top of all commits in arrow-rs, so that history of arrow-rs is preserved.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          path: /github/home/target
+          # these represent compiled steps of both dependencies and arrow
+          # and thus are specific for a particular OS, arch and rust version.
+          key: ${{ runner.os }}-amd64-target-cache-stable
+      - name: Setup parquet files
+        run: |
+          apt update && apt install python3-pip python3-venv -y -q
+          python3 -m venv venv
+          venv/bin/pip install pip --upgrade
+          venv/bin/pip install pyarrow==3
```
Comment (jimexist) [inline .github/workflows/test.yml:35]: use 4.0.1? https://pypi.org/project/pyarrow/

**Comment by jorgecarleitao** [comment]
@elferherrera @jhorstmann , for Apache's IP clearance:

* would it be ok for your contributions to parquet2 to be donated to Apache under the CLA?
* if yes, could you sign and submit a CLA, so that we have the paperwork in order for this to proceed?

Sorry for the inconvenience, this does not seem to be needed in normal contributions, but for some reason for these clearances it is required.

**Comment by jhorstmann** [comment]
I'm of course ok with the donation, will check the CLA requirements later

**Comment by jorgecarleitao** [comment]
@jhorstmann , did you submitted a CLA?

**Pull request closed without merging**